src/components/Table.tsx

```diff
@@ -0,0 +1,101 @@
+/** Load dependencies */
+import React from "react";
+
+/** Load types */
+import { Metric, Types, Title } from "../types";
+
+/** Interfaces */
+interface TableProps {
+  titles: Title[];
+  data: Metric[];
+  toggleVisibility: (position: number) => void;
+  sort?: (position: number) => void;
+}
+
+/** Component */
+const Table: React.FunctionComponent<TableProps> = ({
+  titles,
+  data,
+  toggleVisibility,
+  sort,
+}: TableProps) => {
+  let copyData = data;
+
+  const valueType = (value: number, type: Types) => {
+    switch (type) {
+      case "percentage":
+        return `${value}%`;
+      case "hours":
+        return `${value} hours`;
+      case "secs":
+        return `${value} secs`;
+      default:
+        return value;
+    }
+  };
+
+  return (
+    <div className="box">
+      <table className="table is-fullwidth">
+        <thead>
+          <tr>
+            {titles.map((eachTitle, position) => {
+              let sortIcon;
+              switch (eachTitle.sort?.state) {
+                case "asc":
+                  sortIcon = "mdi-sort-ascending";
+                  break;
+                case "desc":
+                  sortIcon = "mdi-sort-descending";
+                  break;
+                default:
+                  sortIcon = "mdi-sort";
+                  break;
+              }
+              return (
+                <th key={eachTitle.label}>
+                  {eachTitle.label}
+                  {sort && eachTitle?.sort?.key && (
+                    <span
+                      className="icon is-clickable"
+                      onClick={() => sort(position)}
+                    >
+                      <i className={`mdi ${sortIcon}`}></i>
+                    </span>
+                  )}
+                </th>
+              );
+            })}
+          </tr>
+        </thead>
+        <tbody>
+          {copyData.map((eachMetric, i) => {
+            const icon =
+              eachMetric.show === undefined || eachMetric.show
+                ? "mdi-eye"
+                : "mdi-eye-off";
+            return (
+              <tr key={eachMetric.id}>
+                <th>{i + 1}</th>
```

Comment:
```suggestion
                <th>{i + 1}</th>
```

That's a nice idea, but how about we make it more readable? Like defining it as a var and naming it accordingly, e.g. `const index = i + 1;`